baybe/utils/basic.py

```diff
@@ -181,3 +182,75 @@ def find_subclass(base: type, name_or_abbr: str, /):
             f"The class name or abbreviation '{name_or_abbr}' does not refer to any "
             f"of the subclasses of '{base.__name__}'."
         )
+
+
+def register_hooks(
+    target: Callable,
+    pre_hooks: Sequence[Callable] | None = None,
+    post_hooks: Sequence[Callable] | None = None,
+) -> Callable:
+    """Register custom hooks with a given target callable.
+
+    The provided hooks need to be "compatible" with the target in the sense that their
+    signatures are aligned, i.e., no problem arises when arguments intended for the
+    target are passed to the hook:
+
+    * Target and hook need to share the same number of parameters (basic requirement)
+    * Their parameters need to have the same names (required for keyword arguments)
+    * Annotations that **may** be present for the hook parameters must match their
+      target counterpart (safety mechanism to prevent unintended argument use)
+    * Parameters that come with defaults for the target also require defaults for the
```

Comment:
do we really need this requirement? Seems better if the wrapper took care of providing th defaults in cases of calls with ommitted default calls
(ideally we make life of hook provider as easy as possible)

Reply:
fixed